ow_faults/src/FaultInjector.cpp

```diff
@@ -75,6 +84,35 @@ void FaultInjector::setFaultsMessage(ow_faults::PTFaults& msg, ComponentFaults v
   msg.value = static_cast<uint>(value); //should be HARDWARE for now
 }
 
+void FaultInjector::publishAntennaeFaults(const std_msgs::Float64& msg, bool encoder, bool torque, float& m_faultValue, ros::Publisher m_publisher){
+std_msgs::Float64 out_msg;
+
+  if (encoder || torque) {
+    if (isnan(m_faultValue)){
+      m_faultValue = msg.data;
+    }
+    out_msg.data = m_faultValue;
+    m_publisher.publish(out_msg);
+  } else {
+    m_publisher.publish(msg);
+    m_faultValue = msg.data;
+  }
```

Comment:
Simplify
```suggestion
  if (!(encoder || torque)) {
    m_faultValue = msg.data;
  }
  out_msg.data = m_faultValue;
  m_publisher.publish(out_msg);
```

The `isnan` is never is actually executed unless your code can actually starts with faults enabled on init